<h1>ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>


<h2>Zairesh Castillo<h2>

In [1]:
#LIBRARIES

In [2]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [3]:
penguin = pd.read_csv('datasets\\penguins.csv')

In [4]:
penguin

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    MALE  
1         3800.0  FEMALE  
2         3250.0  FEMALE  
3            NaN     NaN  
4         3450.0  FEMALE  
..           ...     ...  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE  

[344 rows x 7 columns]

DESCRIBE ANALYTICS

In [5]:
penguin.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

Filtering a column/data

In [6]:
penguin['species'] == "adelie"

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [7]:
penguin['sex'].mode()

0    MALE
Name: sex, dtype: object

In [8]:
penguin['island'] == 'dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: island, Length: 344, dtype: bool

In [9]:
penguin['species'] == 'adelie'

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [10]:
adelie_df = penguin[penguin['species'] == "adelie"]

In [11]:
adelie_df['body_mass_g'].mean()

nan

In [12]:
penguin['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [13]:
penguin[penguin['species'] == "adelie"]

Empty DataFrame
Columns: [species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex]
Index: []

In [14]:
penguin[penguin['species'] == "adelie"]['body_mass_g'].mean()

nan

In [15]:
penguin['body_mass_g']

0      3750.0
1      3800.0
2      3250.0
3         NaN
4      3450.0
        ...  
339       NaN
340    4850.0
341    5750.0
342    5200.0
343    5400.0
Name: body_mass_g, Length: 344, dtype: float64

In [16]:
penguin['body_mass_g'].mean()

4201.754385964912

In [17]:
penguin['body_mass_g'] >= 5400.0

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341     True
342    False
343     True
Name: body_mass_g, Length: 344, dtype: bool

<h2>Cleaning the Data Frame</h2>

In [18]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

In [19]:
penguin['bill_length_mm']

0      39.1
1      39.5
2      40.3
3       NaN
4      36.7
       ... 
339     NaN
340    46.8
341    50.4
342    45.2
343    49.9
Name: bill_length_mm, Length: 344, dtype: float64

<h1>1. How many Island are there?</h1>

In [20]:
penguin['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3>insigth no.1</h3>

<h4>there are 3 island</h4>

<h1>2. how many female penguin are there in  biscoe?</h1>

In [21]:
penguin.groupby(['sex','island']).count()

species  bill_length_mm  bill_depth_mm  flipper_length_mm  \
sex    island                                                                 
FEMALE Biscoe          80              80             80                 80   
       Dream           61              61             61                 61   
       Torgersen       24              24             24                 24   
MALE   Biscoe          83              83             83                 83   
       Dream           62              62             62                 62   
       Torgersen       23              23             23                 23   

                  body_mass_g  
sex    island                  
FEMALE Biscoe              80  
       Dream               61  
       Torgersen           24  
MALE   Biscoe              83  
       Dream               62  
       Torgersen           23

<h3>insight no.2</h3>

<h4>there are 80 female penguin in biscoe</h4>

<h1>3. how many male penguin are there in torgersen?<h1/> 

In [22]:
penguin.groupby(['sex','island']).count()

species  bill_length_mm  bill_depth_mm  flipper_length_mm  \
sex    island                                                                 
FEMALE Biscoe          80              80             80                 80   
       Dream           61              61             61                 61   
       Torgersen       24              24             24                 24   
MALE   Biscoe          83              83             83                 83   
       Dream           62              62             62                 62   
       Torgersen       23              23             23                 23   

                  body_mass_g  
sex    island                  
FEMALE Biscoe              80  
       Dream               61  
       Torgersen           24  
MALE   Biscoe              83  
       Dream               62  
       Torgersen           23

<h3>insight no.3</h3>

<h4>there are 23 male penguin in torgersen</h4>

<h1>4. which island has the most penguin?</h1>

In [23]:
penguin.groupby(['island']).count()

species  bill_length_mm  bill_depth_mm  flipper_length_mm  \
island                                                                 
Biscoe         168             167            167                167   
Dream          124             124            124                124   
Torgersen       52              51             51                 51   

           body_mass_g  sex  
island                       
Biscoe             167  163  
Dream              124  123  
Torgersen           51   47

<h3>insight no.4</h3>

<h4>the island that has the most penguin is biscoe </h4>

<h1>5. which species is the heaviest?</h1>

In [24]:
penguin.groupby(['species'])['body_mass_g'].mean()

species
Adelie       3700.662252
Chinstrap    3733.088235
Gentoo       5076.016260
Name: body_mass_g, dtype: float64

<h3>insight no.5</h3>

<h4>the heviest species is chinstrap</h4>

<h1>6.which species has the largest bill_depth?</h1>

In [25]:
penguin.groupby(['species'])['bill_depth_mm'].mean()

species
Adelie       18.346358
Chinstrap    18.420588
Gentoo       14.982114
Name: bill_depth_mm, dtype: float64

<h3>insight no.6</h3>

<h4>the species with the largest bill depth is chinstrap</h4>

<h1>7. which gender has the most number of penguin<h1/>

In [26]:
penguin.groupby(['sex',]).count()                                                                                                       

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
sex                                                                         
FEMALE      165     165             165            165                165   
MALE        168     168             168            168                168   

        body_mass_g  
sex                  
FEMALE          165  
MALE            168

<h3>insight no.7<h3/>
<h4>the gender that has the most number of penguin is male<h4/>

<h1>8. what is the shortest recorded flipper length in datasets?</h1>                                                           

In [28]:
penguin.groupby(['species'])['flipper_length_mm'].mean()

species
Adelie       189.953642
Chinstrap    195.823529
Gentoo       217.186992
Name: flipper_length_mm, dtype: float64

<h3>insight no.8</h3>

<h4>the shortest recorded flipper length is 189.953642</h4>